microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsWorkspaceSymbolParticipant.java

```diff
@@ -57,196 +45,90 @@ public class JaxRsWorkspaceSymbolParticipant implements IJavaWorkspaceSymbolsPar
+			}
+			if (monitor.isCanceled()) {
 				return;
 			}
-		} catch (JavaModelException e) {
-			return;
 		}
 
-		String endpointUri = getEndpointUriFromAnnotatable(annotatable, applicationPrefix);
-		String httpMethods = getHttpMethodsFromAnnotatable(annotatable);
-		if (endpointUri == null || endpointUri.isEmpty() || httpMethods.isEmpty()) {
-			return;
-		}
-
-		Location location = getLocationFromAnnotatable(method, utils);
-		if (location == null) {
-			return;
-		}
-
-		StringBuilder nameBuilder = new StringBuilder("@");
-		nameBuilder.append(endpointUri);
-		nameBuilder.append(": ");
-		nameBuilder.append(httpMethods);
-
-		SymbolInformation symbol = new SymbolInformation();
-		symbol.setName(nameBuilder.toString());
-		symbol.setKind(SymbolKind.Method);
-		symbol.setLocation(location);
-
-		symbols.add(symbol);
-	}
-
-	private static Location getLocationFromAnnotatable(IMethod method, IJDTUtils utils) {
-		try {
-			String uri = method.getResource().getLocationURI().toString();
-			ISourceRange r = method.getNameRange();
-			Range range = utils.toRange(method.getOpenable(), r.getOffset(), r.getLength());
-			if (range == null) {
-				return null;
+		methodsInfo.forEach(methodInfo -> {
+			try {
+				symbols.add(createSymbol(methodInfo, utils));
+			} catch (Exception e) {
+				LOGGER.log(Level.WARNING, "failed to create workspace symbol for jax-rs method", e);
 			}
-			return new Location(uri, range);
-		} catch (JavaModelException e) {
-			return null;
-		}
+		});
 	}
 
-	private static String getEndpointUriFromAnnotatable(IAnnotatable annotatable, String applicationPrefix) {
-		StringBuilder builder = new StringBuilder();
-		IAnnotatable current = annotatable;
-
-		while (current != null && current != (IAnnotatable) ((IJavaElement) current).getAncestor(IJavaElement.TYPE)) {
-			if (hasRestClientAnnotation(current)) {
-				// This is not an endpoint, it's an interface to an outside service
-				return null;
+	/**
+	 * Returns the provider that can provide JAX-RS method info for the given class,
+	 * or null if no provider can provide info.
+	 *
+	 * @param typeRoot the class to collect JAX-RS method info for
+	 * @param monitor the progress monitor
+	 * @return the provider that can provide JAX-RS method info for the given class,
+	 *         or null if no provider can provide info
+	 */
+	private static IJaxRsInfoProvider getProviderForType(ITypeRoot typeRoot, IProgressMonitor monitor) {
+		for (IJaxRsInfoProvider provider : JavaFeaturesRegistry.getInstance().getJaxRsInfoDefinitions()) {
+			if (provider.canProvideJaxRsMethodInfoForClass(typeRoot, monitor)) {
+				return provider;
 			}
-			prependPathSegment(builder, current);
-			current = (IAnnotatable) ((IJavaElement) current).getAncestor(IJavaElement.TYPE);
 		}
-		if (hasRestClientAnnotation(current)) {
-			return null;
+		IJaxRsInfoProvider defaultProvider = DefaultJaxRsInfoProvider.getInstance();
```

Comment:
To avoid having specific code with DefaultJaxRsInfoProvider here, I will add `new DefaultJaxRsInfoProvider()` at the end of the list of getJaxRsInfoDefinitions once all IJaxRsInfoProviders are loaded by extension point.